app/controllers/application_controller.rb

```diff
@@ -17,6 +17,11 @@ def health_check
     render plain: health_check_haiku, status: 200
   end
 
+  def robots
+    robots = File.read(Rails.root + "config/robots.#{Settings.robots}.txt")
+    render plain: robots
+  end
```

Comment:
this is not what `ApplicationController` is for. it gets inherited by every other controller, so you're adding this controller action to every controller in the repo. This is what `HomeController` is for.